tendermint/src/v0_37/abci/request.rs

```diff
@@ -0,0 +1,223 @@
+    BeginBlock(BeginBlock),
+    #[doc = include_str!("../../abci/doc/request-checktx.md")]
+    CheckTx(CheckTx),
+    #[doc = include_str!("../../abci/doc/request-delivertx.md")]
+    DeliverTx(DeliverTx),
+    #[doc = include_str!("../../abci/doc/request-endblock.md")]
+    EndBlock(EndBlock),
+    #[doc = include_str!("../../abci/doc/request-commit.md")]
+    Commit,
+    #[doc = include_str!("../../abci/doc/request-listsnapshots.md")]
+    ListSnapshots,
+    #[doc = include_str!("../../abci/doc/request-offersnapshot.md")]
+    OfferSnapshot(OfferSnapshot),
+    #[doc = include_str!("../../abci/doc/request-loadsnapshotchunk.md")]
+    LoadSnapshotChunk(LoadSnapshotChunk),
+    #[doc = include_str!("../../abci/doc/request-applysnapshotchunk.md")]
+    ApplySnapshotChunk(ApplySnapshotChunk),
+    #[doc = include_str!("../../abci/doc/request-prepareproposal.md")]
+    PrepareProposal(PrepareProposal),
+    #[doc = include_str!("../../abci/doc/request-processproposal.md")]
+    ProcessProposal(ProcessProposal),
+}
+
+impl Request {
+    /// Get the method kind for this request.
+    pub fn kind(&self) -> MethodKind {
+        use Request::*;
+        match self {
+            Flush => MethodKind::Flush,
+            InitChain(_) => MethodKind::Consensus,
+            BeginBlock(_) => MethodKind::Consensus,
+            DeliverTx(_) => MethodKind::Consensus,
+            EndBlock(_) => MethodKind::Consensus,
+            Commit => MethodKind::Consensus,
+            PrepareProposal(_) => MethodKind::Consensus,
+            ProcessProposal(_) => MethodKind::Consensus,
+            CheckTx(_) => MethodKind::Mempool,
+            ListSnapshots => MethodKind::Snapshot,
+            OfferSnapshot(_) => MethodKind::Snapshot,
+            LoadSnapshotChunk(_) => MethodKind::Snapshot,
+            ApplySnapshotChunk(_) => MethodKind::Snapshot,
+            Info(_) => MethodKind::Info,
+            Query(_) => MethodKind::Info,
+            Echo(_) => MethodKind::Info,
+        }
+    }
+}
+
+impl From<ConsensusRequest> for Request {
+    fn from(req: ConsensusRequest) -> Self {
+        match req {
+            ConsensusRequest::InitChain(x) => Self::InitChain(x),
+            ConsensusRequest::BeginBlock(x) => Self::BeginBlock(x),
+            ConsensusRequest::DeliverTx(x) => Self::DeliverTx(x),
+            ConsensusRequest::EndBlock(x) => Self::EndBlock(x),
+            ConsensusRequest::Commit => Self::Commit,
+        }
+    }
+}
+
+impl TryFrom<Request> for ConsensusRequest {
+    type Error = Error;
+    fn try_from(req: Request) -> Result<Self, Self::Error> {
+        match req {
+            Request::InitChain(x) => Ok(Self::InitChain(x)),
+            Request::BeginBlock(x) => Ok(Self::BeginBlock(x)),
+            Request::DeliverTx(x) => Ok(Self::DeliverTx(x)),
+            Request::EndBlock(x) => Ok(Self::EndBlock(x)),
+            Request::Commit => Ok(Self::Commit),
+            _ => Err(Error::invalid_abci_request_type()),
```

Comment:
The new `PrepareProposal` and `ProcessProposal` variants are missing from `ConsensusRequest`.

Reply:
Here are the ones I found: https://github.com/informalsystems/tendermint-rs/compare/mikhail/multi-tc-version-support...aakoshh:tendermint-rs:mikhail/multi-tc-version-support

Reply:
Thank you!